src/misc/support.cpp

```diff
@@ -331,3 +331,15 @@ std::string safe_strerror(int err) noexcept
 	return buf;
 #endif
 }
+
+void set_thread_name(std::thread& thread, const char *name)
+{
+#if HAVE_PTHREAD_SETNAME_NP
+	pthread_t handle = thread.native_handle();
+	pthread_setname_np(handle, name);
+#else
+	// Do-nothing if platform can't name threads
+	(void)thread;
+	(void)name;
```

Comment:
Remove these lines and use mark the parameters with `MAYBE_UNUSED` instead. This way it will be easier for me to mechanically replace `MAYBE_UNUSED` with C++17 `[[maybe_unused]]` in the future.

Clang is silly in warning about using C++17 extensions (we specifically ask compiler if attributes are supported before using them).

Reply:
Got it. Yeah; I didn't know which approach to use given in some cases the variables are 100% used and in others they're not.  Lol.. I suppose that's the meaning of `MAYBE_UNUSED` :smile: